vpr/src/route/build_switchblocks.cpp

```diff
@@ -422,6 +432,73 @@ static bool is_core(const DeviceGrid& grid, int x, int y) {
     return is_core;
 }
 
+static bool match_sb_xy(const DeviceGrid& grid, int x, int y, int sb_x, int sb_y, const e_sb_loc_spec& sb_reg_x, const e_sb_loc_spec& sb_reg_y){
+    
+    //if one of sb_x and sb_y is defined, we either know the exact location (x,y) or the exact x location (will apply it to all columns) 
+    //or the exact y location (will apply it to all rows)   
+    if(sb_x != -1 || sb_y != -1){
+        if(x == sb_x && y == sb_y){
+            return true;
+        } 
+
+        if(x == sb_x && sb_y == -1){
+            return true;
+        }
+        
+        if(sb_x == -1 && y == sb_y){
+            return true;
+        }
+    }
+
+    //if both sb_x and sb_y is not defined, we have a region that we should apply this SB pattern to, we just need to check
+    //whether the location passed into this function falls within this region or not
+    if(sb_x == -1 && sb_y == -1){
+        //calculate the appropiate region based on the repeatx/repeaty and current location.
+        //This is to determine whether the given location is part of the current SB specified region with regular expression or not
+        //After region calculation, the current SB will apply to this location if:
+        // 1) the given (x,y) location falls into the calculated region 
+        // *AND*
+        // 2) incrx/incry are respected within the region, this means all locations within the calculated region is
+        //    is not necessary crossponds to the current SB. If incrx/incry is equal to 1, then all locations within the 
+        //    calculated region is valid. 
```

Comment:
a few typos
is is --> do
necessary --> necessarily
crossponds -- > correspond
is valid --> are valid